Patches/Rimworld - Altered Carbon/ThingDefs_Weapons.xml

```diff
@@ -0,0 +1,366 @@
+
+          <value>
+            <li Class="CombatExtended.GunDrawExtension">
+              <DrawSize>0.80,0.80</DrawSize>
+              <DrawOffset>0.0,-0.08</DrawOffset>
+            </li>
+          </value>
+        </li>
+
+        <!-- === Shock PDW === -->
+        <li Class="CombatExtended.PatchOperationMakeGunCECompatible">
+          <defName>AC_Gun_ShockPDW</defName>
+
+          <statBases>
+            <Mass>3.2</Mass>
+            <Bulk>5.00</Bulk>
+            <SwayFactor>1.04</SwayFactor>
+            <ShotSpread>0.09</ShotSpread>
+            <SightsEfficiency>2</SightsEfficiency>
+            <RangedWeapon_Cooldown>0.36</RangedWeapon_Cooldown>
+          </statBases>
+
+          <Properties>
+            <recoilAmount>1.30</recoilAmount>
+            <verbClass>CombatExtended.Verb_ShootCE</verbClass>
+            <hasStandardCommand>True</hasStandardCommand>
+            <defaultProjectile>Bullet_6x24mmCharged</defaultProjectile>
+            <burstShotCount>6</burstShotCount>
+            <ticksBetweenBurstShots>6</ticksBetweenBurstShots>
+            <warmupTime>0.6</warmupTime>
+            <range>46</range>
+            <soundCast>Shot_HeavySMG</soundCast>
+            <soundCastTail>GunTail_Heavy</soundCastTail>
+            <muzzleFlashScale>9</muzzleFlashScale>
+          </Properties>
+
+          <AmmoUser>
+            <magazineSize>30</magazineSize>
+            <reloadTime>4</reloadTime>
+            <ammoSet>AmmoSet_6x24mmCharged</ammoSet>
+          </AmmoUser>
+
+          <FireModes>
+            <aiAimMode>AimedShot</aiAimMode>
+            <aiUseBurstMode>True</aiUseBurstMode>
+            <aimedBurstShotCount>3</aimedBurstShotCount>
+          </FireModes>
+
+          <weaponTags>
+            <li>SpacerGun</li>
+            <li>AC_LateIndustrialGun</li>
+          </weaponTags>
+        </li>
+
+        <!-- === GunDrawExtension === -->
+        <li Class="PatchOperationAddModExtension">
+          <xpath>/Defs/ThingDef[defName="AC_Gun_ShockPDW"]</xpath>
+
+          <value>
+            <li Class="CombatExtended.GunDrawExtension">
+              <DrawSize>0.88,0.88</DrawSize>
+              <DrawOffset>0.0,-0.01</DrawOffset>
+            </li>
+          </value>
+        </li>
+
+        <!-- === MAC Rifle === -->
+        <li Class="CombatExtended.PatchOperationMakeGunCECompatible">
+          <defName>AC_Gun_MACRifle</defName>
+
+          <statBases>
+            <Mass>5.40</Mass>
+            <Bulk>11.10</Bulk>
+            <SwayFactor>1.07</SwayFactor>
+            <ShotSpread>0.04</ShotSpread>
+            <SightsEfficiency>2</SightsEfficiency>
+            <RangedWeapon_Cooldown>0.36</RangedWeapon_Cooldown>
+          </statBases>
+
+          <Properties>
+            <verbClass>CombatExtended.Verb_ShootCE</verbClass>
+            <hasStandardCommand>True</hasStandardCommand>
+            <defaultProjectile>Bullet_76x34mmRailgun</defaultProjectile>
+            <warmupTime>1.5</warmupTime>
```

Comment:
Are the stats being calculated for a sniper rifle or an AMR? Because I think the latter may be more appropriate, since you're what is basically a .50 cal BMG round. I'd increase warmup, cooldown, or both, otherwise I worry your rate of fire is going to be too high, especially with a laser-beam accurate weapon and projectile.

Reply:
The rifle is technically between .50 BMG and 14.5x114m Soviet so you're right i should probably change some numbers